### PR TITLE
Extend unit tests for BMP & Datasets 

### PR DESCRIPTION
**Pull request by piponazo**
In this PR I am continuing with the addition of unit tests for some parts of the code that I am analysing. I try to focus on some areas of the code which were not covered by either unit tests or the python tests. 

NOTE: I am also opening the PR as DRAFT so that I can check what is going wrong with the codecov reports. I'll do some modifications there aiming to get more stable results. 
UPDATE:  I found out that the difference between codecov (61.49%) and the local coverage reports (75.1%) is due to the different treatment of partial coverage. While locally a partial coverage is treated as a HIT, in codecov is treated as a MISS:  https://docs.codecov.com/docs/frequently-asked-questions#how-is-coverage-calculated

### CODE REVIEW

**Review by hassec — DISMISSED** [review]
Had a brief look and I think this all looks good :+1:   

really happy to see more testing! :blush:

File: include/exiv2/bmpimage.hpp [hunk 1]
```diff
@@ -84,20 +84,11 @@ namespace Exiv2 {
               yet(?) implemented. Calling it will throw an Error(kerWritingImageFormatUnsupported).
          */
         void writeMetadata() override;
-        /*!
-          @brief Todo: Not supported yet(?). Calling this function will throw
-              an instance of Error(kerInvalidSettingForImage).
```
Comment (hassec) [inline include/exiv2/bmpimage.hpp:89]: Seems like this function remains not implemented, so I don't think this comment was a bad idea, but don't have strong feeling either way.

Comment (piponazo) [reply include/exiv2/bmpimage.hpp:89]: I do not know what I was thinking when I deleted these comments ... I took a look to the BMP format and it seems like EXIF, IPTC are not really supported by BMP. But anyways, it makes sense to indicate what the methods can throw. I will recover these comments :+1:

**Review by hassec — APPROVED** [review]
LGTM :+1:

**Pull request merged**